pyigm/fN/fnmodel.py

```diff
@@ -0,0 +1,456 @@
+""" Class for f(N) modeling
+"""
+from __future__ import print_function, absolute_import, division, unicode_literals
+
+import numpy as np
+import os, imp
+from scipy import interpolate as scii
+import pdb
+
+from astropy.io import fits
+from astropy.utils.misc import isiterable
+from astropy import units as u
+from astropy import constants as const
+
+
+from pyigm import utils as pyigmu
+
+
+# Path for pyigm
+pyigm_path = imp.find_module('pyigm')[1]
+
+
+class FNModel(object):
+    """A Class for f(N,X) models
+
+    Attributes
+    ----------
+    mtype : str
+      Model type for the fN
+      'Hspline' -- Hermite monotonic spline
+      'Gamma' -- Gamma function [Following Inoue+14]
+    zmnx : tuple
+      Redshift range where this model applies (zmin,zmax)
+    pivots : array
+          log NHI values for the pivots
+    param : ndarray
+      Parameter array
+    zpivot : float, optional
+          Pivot for redshift evolution (2.4)
+    gamma : float, optional
+          Power law for dN/dX, not dN/dz (1.5)
+    """
+    @classmethod
+    def default_model(cls, use_mcmc=False):
+        """ Pass back a default fN_model from Prochaska+14
+
+        Tested against XIDL code by JXP on 09 Nov 2014
+
+        Parameters
+        ----------
+        recalc : boolean, optional (False)
+          Recalculate the default model
+        use_mcmc : boolean, optional (False)
+          Use the MCMC chain to generate the model
+        write : boolean, optional (False)
+          Write out the model
+        """
+        if use_mcmc:
+            # MCMC Analysis (might put these on a website)
+            chain_file = (os.environ.get('DROPBOX_DIR')+
+                        'IGM/fN/MCMC/mcmc_spline_k13r13o13n12_8.fits.gz')
```

Comment:
I think this is the only one and it is not intended to be used by anyone but me.

The others, e.g.             fN_file = (pyigm_path+'/data/fN/fN_spline_z24.fits.gz')
come with pyigm.